tardis/workflows/v_inner_solver.py

```diff
@@ -0,0 +1,341 @@
+            shape=(self.simulation_state.geometry.no_of_shells,),
+        )
+        a2_expanded = np.empty_like(a1_expanded)
+
+        a1_expanded[m1] = a1
+        a2_expanded[m2] = a2
+
+        joint_mask = m1 & m2
+
+        return a1_expanded[joint_mask], a2_expanded[joint_mask]
+
+    def check_convergence(
+        self,
+        estimated_values,
+    ):
+        """Check convergence status for a dict of estimated values
+
+        Parameters
+        ----------
+        estimated_values : dict
+            Estimates to check convergence
+
+        Returns
+        -------
+        bool
+            If convergence has occurred
+        """
+        convergence_statuses = []
+
+        mask = estimated_values["mask"]
+
+        if not np.all(mask == self.property_mask):
+            convergence_statuses.append(False)
+            # Need to set status to False if change in mask size
+            logger.info(f"Resized Geometry, Convergence Suppressed")
+
+        for key, solver in self.convergence_solvers.items():
+
+            current_value = getattr(self.simulation_state, key)
+            estimated_value = estimated_values[key]
+
+            if key in ["t_radiative", "dilution_factor"]:
+                current_value, estimated_value = self.reproject(
+                    current_value, self.property_mask, estimated_value, mask
+                )
+                no_of_shells = current_value.shape[0]
+            else:
+                no_of_shells = 1
+
+            convergence_statuses.append(
+                solver.get_convergence_status(
+                    current_value, estimated_value, no_of_shells
+                )
+            )
+
+        if np.all(convergence_statuses):
+            hold_iterations = self.convergence_strategy.hold_iterations
+            self.consecutive_converges_count += 1
+            logger.info(
+                f"Iteration converged {self.consecutive_converges_count:d}/{(hold_iterations + 1):d} consecutive "
+                f"times."
+            )
+
+            return self.consecutive_converges_count == hold_iterations + 1
+
+        self.consecutive_converges_count = 0
+        return False
+
+    def solve_simulation_state(self, estimated_values):
+        """Update the simulation state with new inputs computed from previous
+        iteration estimates.
+
+        Parameters
+        ----------
+        estimated_values : dict
+            Estimated from the previous iterations
+
+        Returns
+        -------
+        next_values : dict
+            The next values assigned to the simulation state
+        """
+        next_values = super().solve_simulation_state(estimated_values)
+        self.simulation_state.geometry.v_inner_boundary = next_values[
+            "v_inner_boundary"
+        ]
+        self.simulation_state.blackbody_packet_source.radius = (
+            self.simulation_state.r_inner[0]
+        )
+
+        return next_values
+
+    def solve_radiation_field(self):
```

Comment:
That is what I was thinking.  That or we set the radiation field using the simulation_state.radiation_field_state values since the plasma *always* computes values for every shell